lib/devicetrust/authn/authn.go

```diff
@@ -223,5 +228,25 @@ func (c *Ceremony) authenticateDeviceTPM(
 			TpmChallengeResponse: challengeResponse,
 		},
 	})
-	return trace.Wrap(err)
+	if err != nil {
+		return trace.Wrap(consumeStreamToError(err, stream))
+	}
+	return nil
+}
+
+// consumeStreamToError reads from the gRPC bi-directional stream until an error is encountered.
+// The client only sees the server's error when it tries to read from the stream.
+// This function repeatedly calls Recv to surface the first error encountered.
+// Use this function when the stream's content is irrelevant and only the error matters.
+func consumeStreamToError(sendErr error, stream devicepb.DeviceTrustService_AuthenticateDeviceClient) error {
```

Comment:
I defined this function 3 times to avoid making any of these functions publicly available in `api` in order to avoid people to incorrectly use them